src/bench/mempool_stress.cpp

```diff
@@ -118,5 +199,63 @@ static void MempoolCheck(benchmark::Bench& bench)
     });
 }
 
+static void MemPoolMiningScoreCheck(benchmark::Bench& bench)
+{
+    // Default test: each cluster is of size 20, and we'll try to RBF with a
+    // transaction that merges 10 clusters, evicting 10 transactions from each.
+
+    FastRandomContext det_rand{true};
+    int childTxs = 10;
+    if (bench.complexityN() > 1) {
+        childTxs = static_cast<int>(bench.complexityN());
+    }
+    const auto testing_setup = MakeNoLogFileContext<const TestingSetup>(ChainType::MAIN);
+    CTxMemPool& pool = *testing_setup.get()->m_node.mempool;
+
+    LOCK2(cs_main, pool.cs);
+
+    std::vector<CTransactionRef> parent_txs_for_rbf;
+    std::set<uint256> child_txs_to_conflict_with;
+
+    for (int i=0; i<10; i++) {
+        std::vector<CTransactionRef> transactions = CreateCoinCluster(det_rand, childTxs, /*min_ancestors=*/1);
+        parent_txs_for_rbf.push_back(transactions[0]);
+        // Conflict with everything after the first 10 transactions
+        for (size_t j=10; j<transactions.size(); ++j) {
+            child_txs_to_conflict_with.insert(transactions[j]->GetHash());
+        }
+
+        // Add all transactions to the mempool.
+        for (auto& tx : transactions) {
+            AddTx(tx, pool, det_rand);
+        }
+    }
+
+    // Construct a transaction that spends from each of the parent transactions
+    // selected.
+    CMutableTransaction tx = CMutableTransaction();
+    tx.vin.resize(10);
+    for (size_t i=0; i<parent_txs_for_rbf.size(); ++i) {
+        tx.vin[i].prevout = COutPoint(parent_txs_for_rbf[i]->GetHash(), 0);
+        tx.vin[i].scriptSig = CScript() << i;
+    }
+    tx.vout.resize(1);
+    for (auto& out : tx.vout) {
+        out.scriptPubKey = CScript() << CScriptNum(det_rand.randrange(19)+1) << OP_EQUAL;
+        out.nValue = 10 * COIN;
+    }
+
+    CTxMemPool::setEntries all_conflicts = pool.GetIterSet(child_txs_to_conflict_with);
+    CTxMemPoolEntry entry(MakeTransactionRef(tx), det_rand.randrange(10000)+1000, 0, 1, 0, false, 4, LockPoints());
+
+    bench.run([&]() NO_THREAD_SAFETY_ANALYSIS {
+        CTxMemPool::Limits limits(pool.m_limits);
+        pool.CalculateMiningScoreOfReplacementTx(entry, det_rand.randrange(30000)+1000, all_conflicts, limits);
+    });
+}
+
+BENCHMARK(MemPoolAncestorsDescendants, benchmark::PriorityLevel::HIGH);
+BENCHMARK(MemPoolAddTransactions, benchmark::PriorityLevel::HIGH);
```

Comment:
dd6684ab665bc8ae76b76fdd2e578fc77d562a52: While you're touching this, can you rename `MempoolCheck` to `MemPoolCheck`, `MempoolEviction` to `MemPoolEviction` and `ComplexMemPool` to `MempoolComplex`? That makes `-filter=MemPool.*` work

As a workaround, `-filter=.*Mem.*` does work.